tests/Unit/NumericalAlgorithms/Spectral/Test_Spectral.cpp

```diff
@@ -72,35 +75,78 @@ void test_exact_differentiation(const Function& max_poly_deg) {
   }
 }
 
+template <Spectral::Basis BasisType, Spectral::Quadrature QuadratureType>
+void test_weak_differentiation() {
+  static_assert(BasisType == Spectral::Basis::Legendre,
+                "test_weak_differentiation_matrix may not be correct for "
+                "non-Legendre basis functions.");
+  INFO("Test weak differentiation.");
+  CAPTURE(BasisType);
+  CAPTURE(QuadratureType);
+
+  for (size_t n = Spectral::minimum_number_of_points<BasisType, QuadratureType>;
+       n <= Spectral::maximum_number_of_points<BasisType>; n++) {
+    CAPTURE(n);
+    const DataVector& quad_weights =
+        Spectral::quadrature_weights<BasisType, QuadratureType>(n);
+    const Matrix& weak_diff_matrix =
+        Spectral::weak_flux_differentiation_matrix<BasisType, QuadratureType>(
+            n);
+    const Matrix& diff_matrix =
+        Spectral::differentiation_matrix<BasisType, QuadratureType>(n);
+    Matrix expected_weak_diff_matrix(diff_matrix.rows(), diff_matrix.columns());
+
+    for (size_t i = 0; i < diff_matrix.rows(); ++i) {
+      for (size_t l = 0; l < diff_matrix.rows(); ++l) {
```

Comment:
Here (and a few lines below) did you also want to use the variable `n` as upper bound?